ale_linters/typescript/tslint.vim

```diff
@@ -37,9 +37,12 @@ function! ale_linters#typescript#tslint#Handle(buffer, lines)
     return l:output
 endfunction
 
+let s:tsconfig_path = ale#util#FindNearestFile(bufnr('%'), 'tslint.json')
+let s:tslint_options = empty(s:tsconfig_path) ? '' : '-c ' . s:tsconfig_path
```

Comment:
This runs only once when the file is loaded, so this won't work. You'll need to run this from within a `command_callback`, using the buffer number provided to the function.